src/output-writer/google-sheets-writer/charts.ts

```diff
@@ -0,0 +1,140 @@
+import type Sheets from 'googleapis/build/src/apis/sheets/v4.js';
+
+import type { Column } from '../index.js';
+import { ColumnType } from '../index.js';
+
+import { sheetNames } from './index.js';
+
+export const getCharts = (
+  sheetId: number,
+  auditBreakdownSheetId: number,
+  columns: Column[]
+) => {
+  let chartRow = 0;
+
+  const charts: {
+    column: Column;
+    chartRow: number;
+    columnIndex: number;
+  }[] = [];
+
+  const batchUpdates = columns
+    .map((column, i): Sheets.sheets_v4.Schema$Request | null => {
+      if (
+        column.field.type === ColumnType.AuditScore &&
+        column.field.hasAuditValueColumn
+      )
+        return null;
+
+      const thisChartRow = chartRow;
+      chartRow += 15;
+      const columnIndex = i + 1;
+
+      charts.push({
+        column,
+        chartRow: thisChartRow,
+        columnIndex,
+      });
+
+      return {
+        // Histogram chart for each field
+        addChart: {
+          chart: {
+            position: {
+              overlayPosition: {
+                anchorCell: {
+                  sheetId: auditBreakdownSheetId,
+                  columnIndex: 0,
+                  rowIndex: thisChartRow,
+                },
+                widthPixels: 800,
+                heightPixels: 300,
+              },
+            },
+            spec: {
+              title: `Distribution of ${column.name} (${column.nameDetail})`,
+              histogramChart: {
+                series: [
+                  {
+                    data: {
+                      sourceRange: {
+                        sources: [
+                          {
+                            sheetId,
+                            startRowIndex: 2,
+                            startColumnIndex: columnIndex,
+                            endColumnIndex: columnIndex + 1,
+                          },
+                        ],
+                      },
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        },
+      };
+    })
+    .filter(Boolean as any as <T>(input: T) => input is Exclude<T, null>);
```

Comment:
Can you add a comment here? I'm having trouble understanding this filter. 🔍 👀 🙃

Reply:
It is just `.filter(Boolean)`. TS doesn't know that Boolean is a function that will make it filter out any `null` values. The last part gives it a type that says "this is a function that will return true for non-null values"

Reply:
I can add a comment

Reply:
here's the TS docs for the `(input) => input is ...` syntax: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates